databuilder/extractor/dashboard/tableau/tableau_dashboard_extractor.py

```diff
@@ -39,6 +40,10 @@ def execute(self) -> Iterator[Dict[str, Any]]:
                           if workbook['projectName'] not in
                           self._conf.get_list(TableauGraphQLApiMetadataExtractor.EXCLUDED_PROJECTS, [])]
         base_url = self._conf.get(TableauGraphQLApiMetadataExtractor.TABLEAU_BASE_URL)
+        site_name = self._conf.get_string(TableauGraphQLApiMetadataExtractor.SITE_NAME)
```

Comment:
I was not able to find a concrete answer for this when searching the Tableau docs, but my understanding based on the code below is that unless specifically configured for multiple sites (tenants), the `/site/{site_name}` portion of the URL is not present; that is, for a single-tenant installation, the URLs will still have the form `https://tableau-server/my-site/` so that there is no specific site given and the `site_name` would be `''`. Additionally, the only other place `SITE_NAME` is used is [here](https://github.com/amundsen-io/amundsendatabuilder/blob/dddf7a1f001de979acbd6e5799a0eba54e8d233d/databuilder/extractor/dashboard/tableau/tableau_dashboard_utils.py#L181) during the authentication process, which also expects an empty string for a single-site installation.

Because of this, I think it would be wise to provide a default value of `''` for this call to `conf.get_string` in both of these instances, since a user would not necessarily need to specify a `SITE_NAME` if they only have one site, and failing to specify a value they don't strictly need to provide shouldn't cause a failure. On the other hand, I could also see a case for not providing a default, as one could argue that one does need to specify the config value since it is actually "used", but it just doesn't have any characters inside it. 

Since this potential problem already exists elsewhere in the code outside the scope of this PR, and could be argued against, I wouldn't consider this a blocking issue, but just a proposal. I'm curious to get others thoughts on this issue as well.

Reply:
> my understanding based on the code below is that unless specifically configured for multiple sites (tenants), ...

You are correct. Ref: https://help.tableau.com/current/pro/desktop/en-us/embed_structure.htm

> I think it would be wise to provide a default value of '' for this call to conf.get_string in both of these instances, since a user would not necessarily need to specify a SITE_NAME if they only have one site, and failing to specify a value they don't strictly need to provide shouldn't cause a failure.

Sounds good. I added a default value for a site name.

Speaking only of the `tableau_dashboard_extractor` module, other values such as `base_url` are necessary to work. (cluster name also looks necessary for making dashboard key in a database.) As you said, this is a potential problem, but I'd like others to go out of my PR scope.